src/shim/message.rs

```diff
@@ -10,7 +10,7 @@ use serde::{Deserialize, Serialize};
 
 use crate::shim::{address::Address, econ::TokenAmount};
 
-#[derive(Clone, PartialEq, Eq, Serialize, Deserialize, Debug, Hash)]
+#[derive(Clone, Default, PartialEq, Eq, Serialize, Deserialize, Debug, Hash)]
 #[serde(transparent)]
 pub struct Message(Message_v3);
```

Comment:
Let's bite the bullet and copy the fields of `Message_v3` into our `Message`. It'll have to be done eventually.